### ISSUE TITLE
"Unknown command" when running bash-language-server image

### ISSUE THREAD

**Issue by aidalgol** [aidalgol/dockered-language-servers#5]
@xendk I'm getting an error when I try to run the bash ls image.
```
$ docker run -it bash-ls:latest 
Unknown command '/usr/local/bin/node /usr/local/bin/bash-language-server'. Run with -h for help.
```
Any ideas?

**Comment by aidalgol** [comment]
Oh, I see.  Alpine doesn't include bash out of the box.

**Comment by xendk** [comment]
Doesn't seem to make much difference?

```shell
~/d/c/d/bash ▶ make bash-ls 
docker build --tag=bash-ls:latest --file=Dockerfile .
Sending build context to Docker daemon  4.096kB
Step 1/4 : FROM node:current-alpine
 ---> 6a9367b2c744
Step 2/4 : RUN apk add bash
 ---> Running in 68c4d6b52f09
fetch http://dl-cdn.alpinelinux.org/alpine/v3.11/main/x86_64/APKINDEX.tar.gz
fetch http://dl-cdn.alpinelinux.org/alpine/v3.11/community/x86_64/APKINDEX.tar.gz
(1/4) Installing ncurses-terminfo-base (6.1_p20200118-r4)
(2/4) Installing ncurses-libs (6.1_p20200118-r4)
(3/4) Installing readline (8.0.1-r0)
(4/4) Installing bash (5.0.11-r1)
Executing bash-5.0.11-r1.post-install
Executing busybox-1.31.1-r9.trigger
OK: 9 MiB in 20 packages
Removing intermediate container 68c4d6b52f09
 ---> 11688eef8b48
Step 3/4 : RUN npm install --global bash-language-server
 ---> Running in 3e5d8f516e45
npm WARN deprecated left-pad@1.3.0: use String.prototype.padStart()
npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
npm WARN deprecated har-validator@5.1.5: this library is no longer supported
npm WARN deprecated request-promise-native@1.0.9: request-promise-native has been deprecated because it extends the now deprecated request package, see https://github.com/request/request/issues/3142
npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142

added 115 packages, and audited 116 packages in 6s

3 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities
npm notice 
npm notice New minor version of npm available! 7.3.0 -> 7.15.0
npm notice Changelog: <https://github.com/npm/cli/releases/tag/v7.15.0>
npm notice Run `npm install -g npm@7.15.0` to update!
npm notice 
Removing intermediate container 3e5d8f516e45
 ---> a10e7f1d64e5
Step 4/4 : ENTRYPOINT ["/usr/local/bin/bash-language-server"]
 ---> Running in 7ca03f60922f
Removing intermediate container 7ca03f60922f
 ---> 756bad9db9b6
Successfully built 756bad9db9b6
Successfully tagged bash-ls:latest
~/d/c/d/bash ▶ docker run -it --rm bash-ls:latest
Unknown command '/usr/local/bin/node /usr/local/bin/bash-language-server'. Run with -h for help.
~/d/c/d/bash ▶ 
````

Besides, even without bash installed `docker run -it --rm bash-ls:latest start` will run just fine. I know, an bash language server image without bash seems a bit wrong, but the language server doesn't seem to care.

**Comment by aidalgol** [comment]
Oops...  I must not have run it exactly the same way when I was testing my change.  Guess we have to just use the full base image.

**Comment by aidalgol** [comment]
OK, I think this is fixed as of e9d6e7a8146b5fe21b127c50f187e5dde48d8fd1.

**Comment by xendk** [comment]
Doesn't look like it:

```
~/d/c/d/bash ▶ make bash-ls
docker build --tag=bash-ls:latest --file=Dockerfile .
Sending build context to Docker daemon  4.096kB
Step 1/3 : FROM node:current-slim
current-slim: Pulling from library/node
69692152171a: Pull complete 
15c9a4a75a9a: Pull complete 
4ecb888cb790: Pull complete 
a645b9158d17: Pull complete 
c73bd791c38c: Pull complete 
Digest: sha256:33dd19c7b7bd9f9f74177f24fa409da52884c447a4f3fd66ae7f1608274ffc36
Status: Downloaded newer image for node:current-slim
 ---> 780735774098
Step 2/3 : RUN npm install --global bash-language-server
 ---> Running in b12a65066be8
npm WARN deprecated left-pad@1.3.0: use String.prototype.padStart()
npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
npm WARN deprecated har-validator@5.1.5: this library is no longer supported
npm WARN deprecated request-promise-native@1.0.9: request-promise-native has been deprecated because it extends the now deprecated request package, see https://github.com/request/request/issues/3142
npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142

added 115 packages, and audited 116 packages in 6s

3 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities
npm notice 
npm notice New minor version of npm available! 7.13.0 -> 7.15.0
npm notice Changelog: <https://github.com/npm/cli/releases/tag/v7.15.0>
npm notice Run `npm install -g npm@7.15.0` to update!
npm notice 
Removing intermediate container b12a65066be8
 ---> 0fb6d993cf25
Step 3/3 : ENTRYPOINT ["/usr/local/bin/bash-language-server"]
 ---> Running in 2dc2ba18e51d
Removing intermediate container 2dc2ba18e51d
 ---> a4f235fbb943
Successfully built a4f235fbb943
Successfully tagged bash-ls:latest
~/d/c/d/bash ▶ docker run -it --rm bash-ls:latest
Unknown command '/usr/local/bin/node /usr/local/bin/bash-language-server'. Run with -h for help.
~/d/c/d/bash ▶ make bash-ls
docker build --tag=bash-ls:latest --file=Dockerfile .
Sending build context to Docker daemon  4.096kB
Step 1/3 : FROM node:current-slim
current-slim: Pulling from library/node
69692152171a: Pull complete 
15c9a4a75a9a: Pull complete 
4ecb888cb790: Pull complete 
a645b9158d17: Pull complete 
c73bd791c38c: Pull complete 
Digest: sha256:33dd19c7b7bd9f9f74177f24fa409da52884c447a4f3fd66ae7f1608274ffc36
Status: Downloaded newer image for node:current-slim
 ---> 780735774098
Step 2/3 : RUN npm install --global bash-language-server
 ---> Running in b12a65066be8
npm WARN deprecated left-pad@1.3.0: use String.prototype.padStart()
npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
npm WARN deprecated har-validator@5.1.5: this library is no longer supported
npm WARN deprecated request-promise-native@1.0.9: request-promise-native has been deprecated because it extends the now deprecated request package, see https://github.com/request/request/issues/3142
npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142

added 115 packages, and audited 116 packages in 6s

3 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities
npm notice 
npm notice New minor version of npm available! 7.13.0 -> 7.15.0
npm notice Changelog: <https://github.com/npm/cli/releases/tag/v7.15.0>
npm notice Run `npm install -g npm@7.15.0` to update!
npm notice 
Removing intermediate container b12a65066be8
 ---> 0fb6d993cf25
Step 3/3 : ENTRYPOINT ["/usr/local/bin/bash-language-server"]
 ---> Running in 2dc2ba18e51d
Removing intermediate container 2dc2ba18e51d
 ---> a4f235fbb943
Successfully built a4f235fbb943
Successfully tagged bash-ls:latest
~/d/c/d/bash ▶ docker run -it --rm bash-ls:latest
Unknown command '/usr/local/bin/node /usr/local/bin/bash-language-server'. Run with -h for help.
```

But what are you trying to fix? It's been working fine when supplied the `start` parameter (as lsp does), I believe the the error you see is just an very odd way to say "argument required". And using slim instead of alpine adds 70MiB to the image.

Well:

```
~/d/c/d/bash ▶ docker run -it --entrypoint bash --rm bash-ls:latest
root@9df2a35498ae:/# less /usr/local/bin/bash-language-server 
bash: less: command not found
root@9df2a35498ae:/# more /usr/local/bin/bash-language-server 
#!/usr/bin/env node
/* eslint-disable no-console */

const server = require('../out/index')
const package = require('../package')

const args = process.argv

const start = args.find(s => s == 'start')
const version = args.find(s => s == '-v' || s == '--version')
const help = args.find(s => s == '-h' || s == '--help')

if (start) {
  server.listen()
} else if (version) {
  console.log(`Version is ${package.version}`)
} else if (help) {
  console.log(`
Usage:
  bash-language-server start
  bash-language-server -h | --help
  bash-language-server -v | --version
  `)
} else {
  const command = args.join(' ')
  console.error(`Unknown command '${command}'. Run with -h for help.`)
}
root@9df2a35498ae:/# 
```

Indeed. If it doesn't find `start`, help or version in the arguments, it prints argv. Argv includes the command, which is `/usr/local/bin/node /usr/local/bin/bash-language-server`.

**Comment by aidalgol** [comment]
Oh, so the issue was that it needs to be given a command.  That is, indeed, a confusing error message.
